nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -780,12 +792,13 @@ def configure_optimizers(self):
         else:
             return [self._optimizer], [self._scheduler]
 
-    def compute_consumed_samples(self, global_step):
+    def compute_consumed_samples(self, global_relative_step=0):
         # TODO: this should be a counter self.consumed_samples
         # and updated after every train_step: self.consumed_samples += global_batch_size
         app_state = AppState()
         consumed_samples = (
-            global_step * app_state.data_parallel_size * self.cfg.micro_batch_size * get_num_microbatches()
+            self.init_consumed_samples
+            + global_relative_step * app_state.data_parallel_size * self.cfg.micro_batch_size * get_num_microbatches()
```

Comment:
Why computation here is different from BERT model?

Reply:
This is the one when using pipeline parallel / apex fwd-bwd functions